### PR TITLE
Fix iOS key events in platform views

### PR DESCRIPTION
**Pull request by cbracken**
This PR fixes key events not getting passed through to platform views in iOS (regression introduced here: https://github.com/flutter/engine/pull/20972).  The only change made was calling the `super` method in the event handlers. 

https://github.com/flutter/flutter/issues/74044

This patch is https://github.com/flutter/engine/pull/25365 by @KammererTob rebased to tip of tree.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.
- [X] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

**Comment by cbracken** [comment]
@googlebot I consent.

**Comment by KammererTob** [comment]
@googlebot I consent.

**Review by chinmaygarde — APPROVED** [review]

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -1096,26 +1096,33 @@ - (void)dispatchPresses:(NSSet<UIPress*>*)presses API_AVAILABLE(ios(13.4)) {
...
+  [super pressesChanged:presses withEvent:event];
   if (@available(iOS 13.4, *)) {
     [self dispatchPresses:presses];
   }
 }
 
-- (void)pressesEnded:(NSSet<UIPress*>*)presses withEvent:(UIEvent*)event API_AVAILABLE(ios(9.0)) {
+- (void)pressesEnded:(NSSet<UIPress*>*)presses
+           withEvent:(UIPressesEvent*)event API_AVAILABLE(ios(9.0)) {
+  [super pressesEnded:presses withEvent:event];
```
Comment (fbartho) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm:1117]: Should this have been moved to end-of-method?

Just like `onViewDidDisappear`

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 2]
```diff
@@ -1096,26 +1096,33 @@ - (void)dispatchPresses:(NSSet<UIPress*>*)presses API_AVAILABLE(ios(13.4)) {
...
   if (@available(iOS 13.4, *)) {
     [self dispatchPresses:presses];
   }
 }
 
 - (void)pressesCancelled:(NSSet<UIPress*>*)presses
-               withEvent:(UIEvent*)event API_AVAILABLE(ios(9.0)) {
+               withEvent:(UIPressesEvent*)event API_AVAILABLE(ios(9.0)) {
+  [super pressesCancelled:presses withEvent:event];
```
Comment (fbartho) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm:1125]: Should this have been moved to end-of-method?

Just like `onViewDidDisappear`

**Review by fbartho — COMMENTED** [review]
Hypothetically, the super-call could delete resources that the `dispatchPresses` implicitly relies on.

So when tearing stuff down, you want to teardown system resources after your own resources, just like when building some stuff up, you want to build the system resources before building your own resources.

**Comment by olibanjoli** [comment]
any chance this comes with flutter 2.3.0 release?

**Pull request merged**